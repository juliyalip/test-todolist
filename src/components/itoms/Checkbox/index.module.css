.spanCheck {
  width: 17px;
  height: 17px;
  display: inline-block;
  border-radius: 50%;
  background-color: var(--primary-background-color);
  border: 1px solid var(--input-border-color);
  position: absolute;
  top: 3px;
  left: 0;
  transition: background 250ms ease;
}
.spanCheck:hover {
  border-color: blue;
}
@media screen and (min-width: 768px) {
  .spanCheck {
    width: 20px;
    height: 20px;
  }
}

.spanCheckActive {
  background: linear-gradient(45deg, violet, rgb(8, 184, 242));
}

.label {
  position: relative;
  cursor: pointer;
  margin-right: 7px;
}
@media screen and (min-width: 1200px) {
  .label {
    margin-right: 5px;
  }
}

.checkbox {
  opacity: 0;
}/*# sourceMappingURL=index.module.css.map */