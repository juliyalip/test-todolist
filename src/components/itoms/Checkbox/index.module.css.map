{"version":3,"sources":["index.module.scss","index.module.css"],"names":[],"mappings":"AAAA;EACE,WAAA;EACA,YAAA;EACA,qBAAA;EACA,kBAAA;EACA,iDAAA;EACA,2CAAA;EACA,kBAAA;EACA,QAAA;EACA,OAAA;EACA,iCAAA;ACCF;ADAE;EACC,kBAAA;ACEH;ADCE;EAfF;IAgBI,WAAA;IACA,YAAA;ECEF;AACF;;ADCA;EACE,4DAAA;ACEF;;ADCA;EACE,kBAAA;EACA,eAAA;EACD,iBAAA;ACED;ADAC;EALD;IAME,iBAAA;ECGA;AACF;;ADAA;EACE,UAAA;ACGF","file":"index.module.css"}