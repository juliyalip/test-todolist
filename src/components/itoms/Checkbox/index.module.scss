.spanCheck {
  width: 20px;
  height: 20px;
  display: inline-block;
  border-radius: 50%;
  background-color: white;
  border: 0.5px solid gray;
  position: absolute;
  top: 3px;
  left: 0;
  &:hover {
    border: 0.5px solid blue;
  }
}

.spanCheckActive {
  background: linear-gradient(
    to right,
    rgba(248, 13, 221, 1),
    rgba(40, 73, 237, 1)
  );
}

.label {
  position: relative;
  cursor: pointer;
  margin-left: 10px;
}
