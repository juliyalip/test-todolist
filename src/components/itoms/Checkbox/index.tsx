import Checkmark from '../Checkmark';
import cn from 'classnames';
import style from './index.module.scss';

interface IProps {
  done: boolean;
  onComplited: (id: string) => void;
  id: string;
}

const Checkbox = ({ done, onComplited, id }: IProps) => {
  return (
    <label className={style.label}>
      <input
        type="checkbox"
        checked={done}
        onChange={() => onComplited(id)}
        className={style.checkbox}
      />
      <span
        className={cn(style.spanCheck, {
          [style.spanCheckActive]: done,
        })}
      >
        {done && <Checkmark />}
      </span>
    </label>
  );
};

export default Checkbox;
