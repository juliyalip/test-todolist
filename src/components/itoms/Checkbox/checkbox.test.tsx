import { render, fireEvent } from '@testing-library/react';
import Checkbox from './index';

describe('test checkbox', () => {
  it('prerender correctly', () => {
    const handleComplited = jest.fn();
    render(<Checkbox done={false} onComplited={handleComplited} id="001" />);
    const input = document.querySelector('input');
    const label = document.querySelector('label');
    const span = document.querySelector('span');
    expect(span).toHaveStyle({
      background:
        'linear-gradient(to right, rgba(248, 13, 221, 1), rgba(40, 73, 237, 1)',
    });
    expect(label).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    if (input) fireEvent.change(input);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
});
