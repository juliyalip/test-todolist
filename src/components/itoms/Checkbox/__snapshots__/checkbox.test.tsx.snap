// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test checkbox prerender correctly 1`] = `
<span
  class="spanCheck"
/>
`;
